### PR TITLE
Add initial support for STM32WBA

### PR DESCRIPTION
**Pull request by gautierg-st**
Add initial support for STM32WBA.
Clocks, UART, I2C and SPI are added.
Hello world sample is functional.

Do not merge before merging HAL part.

### CODE REVIEW

File: drivers/clock_control/clock_stm32_ll_wba.c [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+
+	if (IN_RANGE(pclken->bus, STM32_PERIPH_BUS_MIN, STM32_PERIPH_BUS_MAX) == 0) {
+		/* Attemp to toggle a wrong periph clock bit */
+		return -ENOTSUP;
+	}
+
+	reg = (uint32_t *)(DT_REG_ADDR(DT_NODELABEL(rcc)) + pclken->bus);
+	reg_val = *reg;
+	reg_val |= pclken->enr;
+	*reg = reg_val;
```
Comment (erwango) [inline drivers/clock_control/clock_stm32_ll_wba.c:78]: sys_set_bits(DT_REG_ADDR(DT_NODELABEL(rcc)) + pclken->bus,
		     pclken->enr);

File: drivers/clock_control/clock_stm32_ll_wba.c [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+
+	if (IN_RANGE(pclken->bus, STM32_PERIPH_BUS_MIN, STM32_PERIPH_BUS_MAX) == 0) {
+		/* Attemp to toggle a wrong periph clock bit */
+		return -ENOTSUP;
+	}
+
+	reg = (uint32_t *)(DT_REG_ADDR(DT_NODELABEL(rcc)) + pclken->bus);
+	reg_val = *reg;
+	reg_val &= ~pclken->enr;
+	*reg = reg_val;
```
Comment (erwango) [inline drivers/clock_control/clock_stm32_ll_wba.c:100]: sys_clear_bits(DT_REG_ADDR(DT_NODELABEL(rcc)) + pclken->bus,
		       pclken->enr);

File: drivers/clock_control/clock_stm32_ll_wba.c [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+		return err;
+	}
+
+	dt_val = STM32_CLOCK_VAL_GET(pclken->enr) <<
+					STM32_CLOCK_SHIFT_GET(pclken->enr);
+	reg = (uint32_t *)(DT_REG_ADDR(DT_NODELABEL(rcc)) +
+					STM32_CLOCK_REG_GET(pclken->enr));
+	reg_val = *reg;
+	reg_val |= dt_val;
+	*reg = reg_val;
```
Comment (erwango) [inline drivers/clock_control/clock_stm32_ll_wba.c:131]: sys_clear_bits(DT_REG_ADDR(DT_NODELABEL(rcc)) + STM32_CLOCK_REG_GET(pclken->enr),
		       STM32_CLOCK_MASK_GET(pclken->enr) << STM32_CLOCK_SHIFT_GET(pclken->enr));
	sys_set_bits(DT_REG_ADDR(DT_NODELABEL(rcc)) + STM32_CLOCK_REG_GET(pclken->enr),
		     STM32_CLOCK_VAL_GET(pclken->enr) << STM32_CLOCK_SHIFT_GET(pclken->enr));

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/clock_control/clock_stm32_ll_wba.c (hunk 2 above)
Comment (gautierg-st) [reply drivers/clock_control/clock_stm32_ll_wba.c]: Done

**Review by erwango — CHANGES_REQUESTED** [review]

**Review by erwango — APPROVED** [review]

File: drivers/flash/flash_stm32wbax.c [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+	if (icache_enabled) {
+		/* Since i-cache was disabled, this would start the
+		 * invalidation procedure, so wait for completion.
+		 */
+		rc = icache_wait_for_invalidate_complete();
+
+		/* I-cache should be enabled only after the
+		 * invalidation is complete.
+		 */
+		stm32_icache_enable();
```
Comment (FRASTM) [inline drivers/flash/flash_stm32wbax.c:245]: should the cache be enabled if rc != 0 (timeout for examples ?)

**Review by de-nordic — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]

File: boards/arm/nucleo_wba52cg/doc/nucleo_wba52cg.rst [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+#################
+
+Overview
+********
+
+
+Hardware
+********
+
+
```
Comment (erwango) [inline boards/arm/nucleo_wba52cg/doc/nucleo_wba52cg.rst:13]: Please update this part

File: boards/arm/nucleo_wba52cg/doc/nucleo_wba52cg.rst [hunk 6]
```diff
@@ -0,0 +1,108 @@
+.. _nucleo_wba52cg_board:
+
+ST Nucleo WBA52CG
+#################
+
+Overview
+********
+
```
Comment (erwango) [inline boards/arm/nucleo_wba52cg/doc/nucleo_wba52cg.rst:8]: Please update this part

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_wba52cg/doc/nucleo_wba52cg.rst (hunk 5 above)
Comment (gautierg-st) [reply boards/arm/nucleo_wba52cg/doc/nucleo_wba52cg.rst]: Done

File: drivers/flash/flash_stm32wbax.c (hunk 4 above)
Comment (erwango) [reply drivers/flash/flash_stm32wbax.c:245]: Let's keep it that way for now.

**Review by erwango — DISMISSED** [review]

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**